streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -134,9 +134,7 @@ private static FileLock lockStateDirectory(File stateDir, int retry) throws IOEx
             retry--;
             lock = lockStateDirectory(channel);
         }
-        if (lock == null) {
-            channel.close();
-        }
+
         return lock;
```

Comment:
Yes, that's exactly right, I'll add a comment. I'll attempt to address the leak as well. Thanks @ijuma